completions/ssh

```diff
@@ -434,8 +434,11 @@ _comp_cmd_scp__path_esc='[][(){}<>"'"'"',:;^&!$=?`\\|[:space:]]'
 # Complete remote files with ssh.  If the first arg is -d, complete on dirs
 # only.  Returns paths escaped with three backslashes.
 # shellcheck disable=SC2120
+# @param  $1 `no-overescape` if we should use 1 backslash for escaping instead of 3
```

Comment:
Even though I agree that the original escape seems a bit overdoing it, it is the traditional/standard one that's been there for a long time, and it is still the say ssh does it -- rsync is kind of a "guest" here. In that sense I'd prefer a more neutral variable name here.

Also, requiring a positional boolean-like parameter to have a specific value like here feels somewhat unconventional. Making it an option (i.e. dash prefixed, no value required as it's a boolean) would be more in line with rest of the codebase.

Reply:
> I think we should test the new behavior of scp_remote_files too.

How shall I approach that? It calls ssh inside to fetch completions, so simply calling the function as it is now won't work.

Reply:
@scop I realise this is the current change requested that's still blocking. However, now that the latest rsync has a bigger install base, the problem (that this PR would fix) is spreading. Would you consider merging it even without this test, to pull users out of their misery, please?

Reply:
@scop To summarize the conversation, the initial test followed the `LIVE_HOST` approach that one of the `scp` tests used. The problem was that it was non-trivial how to properly set the host for the continuous integration (CI). However, the existing test case turned out to be skipped in the CI. We decided to use an approach that does not require `LIVE_HOST`: the one using mock functions that reproduces the fixed output of the command `ssh` or `rsync`.